From aa1fb703d3d664deb59e08e78c7ccbcdcc58c84a Mon Sep 17 00:00:00 2001
From: Sebastian Reichel <sebastian.reichel@collabora.com>
Date: Fri, 21 Aug 2026 15:41:34 +0200
Subject: [PATCH 111/293] drm/rockchip: dw_dp: Add system PM support

Explicitly suspend the DP controller when the system goes to sleep and
more importantly resume it properly after system sleep exit to recover
USB-C HPD register state, which is lost when the system goes to sleep.

Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>
---
 drivers/gpu/drm/rockchip/dw_dp-rockchip.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/rockchip/dw_dp-rockchip.c b/drivers/gpu/drm/rockchip/dw_dp-rockchip.c
index 770ab042a..c5211578a 100644
--- a/drivers/gpu/drm/rockchip/dw_dp-rockchip.c
+++ b/drivers/gpu/drm/rockchip/dw_dp-rockchip.c
@@ -271,6 +271,7 @@ static int dw_dp_rockchip_runtime_resume(struct device *dev)
 
 static const struct dev_pm_ops dw_dp_pm_ops = {
 	RUNTIME_PM_OPS(dw_dp_rockchip_runtime_suspend, dw_dp_rockchip_runtime_resume, NULL)
+	SET_LATE_SYSTEM_SLEEP_PM_OPS(pm_runtime_force_suspend, pm_runtime_force_resume)
 };
 
 static const struct rockchip_dw_dp_plat_data rk3588_dp_plat_data = {
-- 
2.53.0

